### PR TITLE
Dikla Rikovitch Hotel Octos

### PR DESCRIPTION
**Pull request by diklaaharoni**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I needed to decide how many classes to create. I decided  to create 4 classes that each class will have one responsibility 
Describe a concept that you gained more clarity on as you worked on this assignment. | I understand better how to work with classes
Describe a _nominal test_ that you wrote for this assignment. | 
Describe an _edge case test_ that you wrote for this assignment. | raise an error if room is already reserved
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | i struggled in the beginning and I wrote the test after I wrote the code

### CODE REVIEW

File: lib/period.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  class Period
+
+    attr_reader :check_in, :check_out
+
+    def initialize(check_in, check_out)
+      @check_in = check_in
+      @check_out = check_out
+    end
+
+    def is_valid?
```
Comment (kariabancroft) [inline lib/period.rb:12]: Methods that end in a `?` should return true or false

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes - I like the way you used a separate object to handle some of the date logic. 
Demonstrated loose coupling |Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes mostly - there are only a few methods that could have improved outputs
Wave 1 requirements | Yes
Wave 2 requirements | No
Wave 3 requirements | No
